website/docs/cli/config/config-file.mdx

```diff
@@ -115,12 +115,29 @@ is available at multiple hostnames, use only one of them consistently.
 Terraform Cloud responds to API calls at both its current hostname
 `app.terraform.io`, and its historical hostname `atlas.hashicorp.com`.
 
-### Credentials Helpers
+### Environment Variable Credentials
 
 If you would prefer not to store your API tokens directly in the CLI
-configuration as described in the previous section, you can optionally instruct
-Terraform to use a different credentials storage mechanism by configuring a
-special kind of plugin program called a _credentials helper_.
+configuration, you may use a host-specific environment variable. Environment variable names should
+have the prefix `TF_TOKEN_` added to the domain name, with periods encoded as underscores.
+For example, the value of a variable named `TF_TOKEN_app_terraform_io` will be used as a
+bearer authorization token when the CLI makes service requests to the hostname "app.terraform.io".
+
+Domain names containing non-ASCII characters should be converted to
+their [punycode equivalent](https://www.charset.org/punycode)
+with an ACE prefix when used as a variable name. For example, token credentials
+for 例えば.com must be set in a variable called `TF_TOKEN_xn--r8j3dr99h_com`.
+
+Hyphens are allowed within variable names by certain tools, such as the `env` utility,
+but are not valid POSIX variable names. Therefore, it can be tricky to
+set them with interactive tools such as bash or zsh so you may encode
+hyphens as double underscores. For example, a token for the domain name "café.fr" can
+be set by either `TF_TOKEN_xn--caf-dma_fr` or `TF_TOKEN_xn____caf__dma_fr` if that is more
+convenient. If both are defined, the variable containing hyphens is preferred.
```

Comment:
I'm working on active voice and also just making things clearer by fussing with the way information is presented in the sentences. As usual - please push back on me if I've edited in a way that loses/changes the meaning :-) Thank you so much for working with me on this! I appreciate you adding in these extra details for users.

```suggestion
Some tools like the `env` utility allow hyphens in variable names, but hyphens create invalid POSIX variable names. Therefore, you can encode hyphens as double underscores when you set variables with interactive tools like Bash or Zsh. For example, you can set a token for the domain name "café.fr" as either `TF_TOKEN_xn--caf-dma_fr` or `TF_TOKEN_xn____caf__dma_fr`. If both are defined, Terraform will use the version containing hyphens.
```